src/data/nat/factorial.lean

```diff
@@ -143,4 +146,50 @@ begin
   { exact add_factorial_succ_le_factorial_add_succ i h }
 end
 
+end factorial
+
+section desc_fac
+
+/-- desc_fac n k = (n + k)! / n!, but implemented in a recursive way for calculation.
+This is closely related to `ring_theory.polynomial.pochhammer`, but much less general. -/
+def desc_fac (n : ℕ) : ℕ → ℕ
+| 0 := 1
+| (k + 1) := (n + k + 1) * desc_fac k
+
+@[simp] lemma desc_fac_zero (n : ℕ) : desc_fac n 0 = 1 := rfl
+
+@[simp] lemma zero_desc_fac (k : ℕ) : desc_fac 0 k = k! :=
+begin
+  induction k with t ht, refl,
+  unfold desc_fac, rw [ht, zero_add, nat.factorial_succ]
+end
+
+lemma desc_fac_succ {n k : ℕ} : desc_fac n k.succ = (n + k + 1) * desc_fac n k := rfl
+
+lemma succ_desc_fac (n : ℕ) : ∀ k, (n + 1) * desc_fac n.succ k = (n + k + 1) * desc_fac n k
+| 0 := by rw [add_zero, desc_fac_zero, desc_fac_zero]
+| (k + 1) :=
+begin
+  have : (n + 1) * ((n.succ + k + 1) * desc_fac n.succ k)
+       = (n.succ + k + 1) * ((n + 1) * desc_fac n.succ k), by ac_refl,
+  rw [desc_fac, this, succ_desc_fac k, desc_fac, succ_eq_add_one], ac_refl
+end
```

Comment:
Similarly, 
```suggestion
| (k + 1) := by rw [desc_fac, mul_left_comm, succ_desc_fac, desc_fac, succ_add, ← add_assoc]
```
(note how `.succ` is defeq to ` + 1`)

Reply:
I was thinking of stopping fighting the order it wants to be and make it `n+1+k`, even though I'd never write that in paper; does that seem reasonable? This appears a little in my rewrite of `pochhammer too